### DISCUSSION TITLE
Improve the Control Panel image transformations

### DISCUSSION THREAD

**Discussion by ammannbe** [craftcms/cms#15339] [Ideas] [answered]
We have a huge image library as an external FTP volume. Each time we add new images, setup the library on a new site or something similar, all those images needs to be regenerated which needs some times, so the user can't se the image immediately.

This is the affected HTML code from the control panel:

```html
<div class="element large hasthumb" title="Zuccardi poligonos familia zuccardi argentinien mendoza rotwein - Deutsch"
    data-kind="image" data-filename="my-image.webp"
    data-image-width="235" data-image-height="800" data-peer-file="" data-movable="" data-replaceable=""
    data-editable-image="" data-type="craft\elements\Asset" data-id="83426" data-site-id="1" data-status="enabled"
    data-label="Zuccardi poligonos familia zuccardi argentinien mendoza rotwein"
    data-url="https://ftp.domain.ch/my-folder/my-image.webp"
    data-settings="{&quot;context&quot;:&quot;index&quot;,&quot;size&quot;:&quot;large&quot;,&quot;showStatus&quot;:false,&quot;showThumb&quot;:true,&quot;showLabel&quot;:true,&quot;showDraftName&quot;:true}"
    data-editable="" data-savable="" data-deletable=""
> 
    <div class="elementthumb" data-sizes="120px" data-srcset="https://ftp.domain.ch/my-folder/_35x120_crop_center-center_none/my-image.webp?v=1717769837 120w, https://ftp.domain.ch/my-folder/_70x240_crop_center-center_none/my-image.webp?v=1717769837 240w">
        <img sizes="120px" srcset="https://ftp.domain.ch/my-folder/_35x120_crop_center-center_none/my-image.webp?v=1717769837 120w, https://ftp.domain.ch/my-folder/_70x240_crop_center-center_none/my-image.webp?v=1717769837 240w" alt="">
    </div>
    <div class="label">
        <span class="title">
            <a href="https://ftp.domain.ch/admin/assets/edit/83426-my-image?site=default">
                Zuccardi poligonos familia zuccardi argentinien mendoza rotwein
            </a>
        </span>
    </div>
</div>
```

When I saw the image variations, I was asking myself if it wouldn’t be possible to show the original image as fallback as long as the transformation is not generated yet.
I know about some JS options and workarounds. But maybe there’s a HTML or PHP only possibility somebody else knows about.
 
With that people could see the image immediately but the transformation can still happen in background (e.g. via a cronjob) to optimize later requests.

**Comment by brandonkelly** [comment, marked as answer]
Loading full-size images in thumbnails would have its own issues. It would potentially significantly increase the bandwidth consumed by the control panel, and slow the browser down, depending on the size of the original images.

I’d recommend you look into a different filesystem type than FTP, such as S3. There’s a reason Craft doesn’t come with a native or first party FTP filesystem type: because FTP is going to be incredibly slow.
